src/network/netdev/hsr.c

```diff
@@ -0,0 +1,142 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <net/if.h>
+#include <netinet/in.h>
+#include <linux/if_arp.h>
+
+#include "hsr.h"
+#include "string-table.h"
+
+static const char* const hsr_protocol_table[_NETDEV_HSR_PROTOCOL_MAX] = {
+        [NETDEV_HSR_PROTOCOL_HSR] = "HSR",
+        [NETDEV_HSR_PROTOCOL_PRP] = "PRP",
+};
+
+DEFINE_STRING_TABLE_LOOKUP(hsr_protocol, HsrProtocol);
+DEFINE_CONFIG_PARSE_ENUM(config_parse_hsr_protocol, hsr_protocol, HsrProtocol,
+                         "Failed to parse Protocol=");
+
+static int netdev_hsr_get_iface_indexes(Hsr *hsr, int *indexes) {
+        Link *link = NULL;
```

Comment:
The initialization is not necessary. Please drop it. Also, please move the declaration into the loop.